Rules/UseCorrectCasing.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// UseCorrectCasing: Check if cmdlet is cased correctly.
+    /// </summary>
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class UseCorrectCasing : ConfigurableRule //IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Analyze the script to check if cmdlet alias is used.
+        /// </summary>
+        public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            IEnumerable<Ast> commandAsts = ast.FindAll(testAst => testAst is CommandAst, true);
+
+            // Iterates all CommandAsts and check the command name.
+            foreach (CommandAst commandAst in commandAsts)
+            {
+                string commandName = commandAst.GetCommandName();
+
+                // Handles the exception caused by commands like, {& $PLINK $args 2> $TempErrorFile}.
+                // You can also review the remark section in following document,
+                // MSDN: CommandAst.GetCommandName Method
+                if (commandName == null)
+                {
+                    continue;
+                }
+
+                using (var powershell = System.Management.Automation.PowerShell.Create())
+                {
+                    var getCommand = powershell.AddCommand("Get-Command")
+                        .AddParameter("Name", commandName)
+                        .AddParameter("ErrorAction", "SilentlyContinue");
+
+                    //if (commandName != null)
+                    //{
+                    //    psCommand.AddParameter("CommandType", commandType);
+                    //}
+
+                    var commandInfo = getCommand.Invoke<CommandInfo>().FirstOrDefault();
+                    if (commandInfo != null)
+                    {
```

Comment:
What do we do about things which look like a command, but can't be found by `Get-Command`. Should we attempt to validate those? Is ignoring them as helpful as we could be? Could we suggest `"get-zebra"` is `"Get-Zebra"` (which could be done via simple regex)

Reply:
Changing `get-zebra` to `Get-Zebra` of the command cannot be is a bit assumptuous IMHO, this would be better suited for something like a configurable script-style rule.
The idea of this PR is to only add something small and simple where we are sure. I could do more, I could also fix the casing of parameter names but the idea is to add something small and minimum viable to have value to the user and based on feedback, invest more time in adding more functionality.